cli/flox-rust-sdk/src/providers/services.rs

```diff
@@ -405,4 +581,202 @@ mod tests {
                 command: bar
         "})
     }
+
+    /// A test helper that starts a `process-compose` instance with a given [ProcessComposeConfig].
+    /// The process is stopped when the instance is dropped or [TestProcessComposeInstance::stop]
+    /// is called.
+    struct TestProcessComposeInstance {
+        _temp_dir: TempDir,
+        socket: PathBuf,
+        child: std::process::Child,
+    }
+
+    impl TestProcessComposeInstance {
+        /// Start a `process-compose` instance with the given [ProcessComposeConfig].
+        /// Wait for the socket to appear before returning.
+        ///
+        /// Panics if the socket doesn't appear after 5 tries with backoff.
+        fn start(config: &ProcessComposeConfig) -> Self {
+            let temp_dir = TempDir::new().unwrap();
```

Comment:
blocking: TempDir::new might return a path that is too long for a socket, so use `flox.temp_dir` or `/tmp`. See
https://github.com/flox/flox/blob/14f3c9c1afc5f7f01e9fd96bd967fd587b0a46f6/cli/flox-rust-sdk/src/flox.rs#L297-L298

Reply:
Oh wow tmpdir::new is already too long 🥲
Will find something shorter